src/mmw/js/src/core/opacityControl.js

```diff
@@ -0,0 +1,80 @@
+/*
```

Comment:
Re: the `shim` directory, external libraries that can't be brought in via NPM or need to be browserified go in there. But since we've changed most of the code, I think it's fine here.